Tests/kaas/k8s-version-policy/k8s_version_policy.py

```diff
@@ -0,0 +1,531 @@
+    -h                                 - Output help
+    """)
+
+
+def parse_arguments(argv):
+    """Parse cli arguments from the script call"""
+    config = Config()
+
+    try:
+        opts, args = getopt.gnu_getopt(argv, "c:k:h", ["config", "kubeconfig", "help"])
+    except getopt.GetoptError:
+        raise ConfigException
+
+    for opt in opts:
+        if opt[0] == "-h" or opt[0] == "--help":
+            raise HelpException
+        if opt[0] == "-c" or opt[0] == "--config":
+            config.config_path = opt[1]
+        if opt[0] == "-k" or opt[0] == "--kubeconfig":
+            config.kubeconfig = opt[1]
+
+    return config
+
+
+def setup_logging(config_log):
+
+    logging.config.dictConfig(config_log)
+    loggers = [
+        logging.getLogger(name)
+        for name in logging.root.manager.loggerDict
+        if not logging.getLogger(name).level
+    ]
+
+    for log in loggers:
+        log.setLevel(config_log['level'])
+
+
+def initialize_config(config):
+    """Initialize the configuration for the test script"""
+
+    try:
+        with open(config.config_path, "r") as f:
+            config.logging = yaml.safe_load(f)['logging']
+    except OSError:
+        logger.warning(f"The config file under {config.config_path} couldn't be found, "
+                       f"falling back to the default config.")
+    finally:
+        # Setup logging if the config file with the relevant information could be loaded before
+        # Otherwise, we initialize logging with the included literal
+        setup_logging(config.logging or logging_config)
+
+    if config.kubeconfig is None:
+        raise ConfigException("A kubeconfig needs to be set in order to test a k8s cluster version.")
+
+    return config
+
+
+@dataclass(frozen=True, eq=True, order=True)
+class K8sVersion:
+    major: int
+    minor: int
+    patch: int = 0
+
+    @property
+    def branch(self):
+        """Get the branch of this version, i.e., the version w/o patch level."""
+        return K8sBranch(self.major, self.minor)
+
+    def __str__(self):
+        return f"{self.major}.{self.minor}.{self.patch}"
+
+
+def parse_version(version_str: str) -> K8sVersion:
+    cleansed = version_str.removeprefix("v").strip()
+    try:
+        major, minor, patch = cleansed.split(".")
+        return K8sVersion(int(major), int(minor), int(patch))
+    except ValueError:
+        raise ValueError(f"Unrecognized version format: {version_str}")
+
+
+@dataclass(frozen=True, eq=True, order=True)
+class K8sBranch:
+    """Identifies a release branch of K8s just by major and minor version."""
+
+    major: int
+    minor: int
+
+    def previous(self):
+        if self.minor == 0:
+            # FIXME: this is ugly
+            return self
+        return K8sBranch(self.major, self.minor - 1)
+
+    def __str__(self):
+        return f"{self.major}.{self.minor}"
+
+
+@dataclass(frozen=True)
+class K8sBranchInfo:
```

Comment:
I'm wondering if this class is really necessary, or if the map from K8sBranch to datetime would suffice (the methods `is_supported` and `is_eol` could be inlined, or a subclass of datetime could be used).

Reply:
I would like to keep the class, because IMHO it is more readable that way :)

Reply:
It's definitely very readable, but there are lots of these classes, and this one seems a bit redundant (even more so with the map, where the key is entirely contained in the value). But I'm not too passionate about this. It's a matter of taste.